### PR TITLE
Remove TemplateManager and VSFileXMLSerializer

### PR DESCRIPTION
**Pull request by royfalk**
Purpose:
- What is this pull request trying to do? Remove older code. In this case, my target was VSFileXMLSerializer. 
- What release is this for? The coming one.
- Is there a project or milestone we should apply this to? @BenjamenMeyer 

Template Manager is part of the audio and does not appear to be used at the moment, but it's relatively clean and modern code. Audio works fine without it. I wonder if it was a replacement to current code.
I also had to gut the test file.

Code Changes:
- [Yes] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- One jump had a rendering freeze for 2-3 seconds. Following jump was fine.

### CODE REVIEW

**Review by Loki1950 — APPROVED** [review]
Not sure about the audio potion IIRC kluass's was trying to modernism the audio

**Comment by BenjamenMeyer** [comment]
NOTE: Seems the builds that failed had an issue with Py3 and Boost-Python. Not sure why but that's the build error.

**Comment by stephengtuggy** [comment]
I will try to review this PR this weekend. Can we get the builds passing before then?

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
Blocking until builds pass.

File: engine/CMakeLists.txt [hunk 1]
```diff
@@ -808,9 +808,9 @@ IF (USE_SYSTEM_BOOST)
         MESSAGE("-- Boost_1_67_Or_Later_Result: ${Boost_1_67_Or_Later_Result}")
         MESSAGE("-- USE_PYTHON_3: ${USE_PYTHON_3}")
         IF (Boost_1_67_Or_Later_Result)
-            SET(BOOST_PYTHON_COMPONENT "python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+            SET(BOOST_PYTHON_COMPONENT "python38")
```
Comment (stephengtuggy) [inline engine/CMakeLists.txt:811]: Why tie Boost-Python to a particular Python version, on a particular set of platforms/variants? Please revert this, or find another way to do it. These `python38` changes are probably why the build is breaking.

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
The Boost_Python changes here are probably what is breaking the build.

Apart from that, I'm worried that the positional audio functionality will no longer work. It works for me at present; does it not work for you guys?

Is there some other code elsewhere that takes care of positional audio, that I'm just not seeing?

**Comment by royfalk** [comment]
Sorry. For some reason I got a whole bunch of emails about changes to the packages and none for this PR.
As I noted, this code is not actually used yet.
Now that I've noticed it, I'll try and get to it today or tomorrow.

**Comment by royfalk** [comment]
OK. So build passes but CodeQL is complaining about code I didn't touch. 
I fixed the CMakeList file, so I think we're good to go.

**Review by stephengtuggy — APPROVED** [review]
OK. Looks fine to me at this point.

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**